llama_index/indices/document_summary/retrievers.py

```diff
@@ -0,0 +1,213 @@
+"""Document summary retrievers.
+
+This module contains retrievers for document summary indices.
+
+"""
+
+from llama_index.callbacks.schema import CBEventType
+from llama_index.indices.document_summary.base import GPTDocumentSummaryIndex
+from llama_index.indices.query.schema import QueryBundle
+from llama_index.indices.base_retriever import BaseRetriever
+from typing import Any, List, Optional, Callable, Tuple, Dict
+from llama_index.data_structs.node import Node, NodeWithScore
+from llama_index.prompts.prompts import QuestionAnswerPrompt
+from llama_index.indices.service_context import ServiceContext
+from llama_index.indices.query.embedding_utils import (
+    get_top_k_embeddings,
+)
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+DEFAULT_CHOICE_SELECT_PROMPT_TMPL = (
+    "A list of documents is shown below. Each document has a number next to it along "
+    "with a summary of the document. A question is also provided. \n"
+    "Select the numbers of the documents "
+    "you should consult to answer the question, in order of relevance. \n"
+    "Do not include any documents that are not relevant to the question. \n"
+    "Example format: \n"
+    "Document 1: <summary of document 1>\n"
```

Comment:
As an alternative format, we found asking it to use JSON format a bit more robust (in case summaries end up with new-line characters or things like that).